src/piton/labelers/core.py

```diff
@@ -0,0 +1,381 @@
+        """Returns the (start, end) of the patient timeline
+
+        Returns:
+            Tuple[datetime.datetime, datetime.datetime]: (start, end)
+        """
+        pass
+
+    @abstractmethod
+    def get_labeler_type(self) -> LabelType:
+        """Return what type of labels this labeler returns. See the Label class."""
+        pass
+
+    def apply(self, patients: Sequence[Patient]) -> LabeledPatients:
+        """Applies the `label()` function one-by-one to each Patient in a sequence of Patients
+
+        Args:
+            patients (Sequence[Patient]): A sequence of Patient objcets
+
+        Returns:
+            LabeledPatients: Maps patients to labels
+        """        
+        patients_to_labels: Dict[int, List[Label]] = {}
+        for patient in patients:
+            patients_to_labels[patient.patient_id] = self.label(patient)
+        return LabeledPatients(patients_to_labels, self.get_labeler_type())
+
+class LabeledPatients(MutableMapping):
+    """Responsible for mapping patients to labels
+        Wrapper class around the output of an LF's `apply()` function
+    """
+    def __init__(self, 
+                 patients_to_labels: Dict[int, List[Label]],
+                 labeler_type: LabelType):
+        self.patients_to_labels: Dict[int, List[Label]] = patients_to_labels
+        self.labeler_type: LabelType = labeler_type
+
+    def as_numpy_arrays(self) -> Tuple[np.array, np.array, np.array]:
+        """Convert `patients_to_labels` to a tuple of np.array's, one for each of:
+            Patient ID, Label value, Label time
+        
+        Returns:
+            Tuple[np.array, np.array, np.array]: (Patient IDs, Label values, Label time)
+        """
+        patient_ids: List[int] = []
+        label_values: List[Any] = []
+        label_times: List[datetime.datetime] = []
+        
+        if self.labeler_type in ["boolean", "numerical", "categorical"]:
+            for patient_id, labels in self.patients_to_labels.items():
+                for label in labels:
+                    patient_ids.append(patient_id)
+                    label_values.append(label.value)
+                    label_times.append(label.time)
+        elif self.labeler_type == "survival":
+            raise
+        else:
+            raise ValueError(
+                "Other label types are not implemented yet for this method"
+            )
+        return (
+            np.array(patient_ids),
+            np.array(label_values),
+            np.array(label_times),
+        )
+
+    def as_list_of_label_tuples(self) -> List[Tuple[int, List[Label]]]:
+        """Convert `patients_to_labels` to a list of (patient_id, Label) tuples
+        """        
+        result: List[Tuple[int, Label]] = []
+        for patient_id, labels in self.patients_to_labels.items():
+            for label in labels:
+                result.append((int(patient_id), label))
+        return result
+
+    def save_to_file(self,
```

Comment:
Remove this function as you should be able to just call pickle.dump(labeled_patients_object, fd)

Reply:
We also need to make sure that pickle.load works for this type as you would expect.